component/file_cache/file_cache.go

```diff
@@ -704,107 +739,171 @@ func (fc *FileCache) OpenFile(options internal.OpenFileOptions) (*handlemap.Hand
 	defer flock.Unlock()
 
 	fc.policy.CacheValid(localPath)
-	downloadRequired, fileExists, attr, err := fc.isDownloadRequired(localPath, options.Name, flock)
+
+	attr, err := fc.NextComponent().GetAttr(internal.GetAttrOptions{Name: options.Name})
+	if err != nil {
+		log.Err("FileCache::download : Failed to get attr of %s [%s]", options.Name, err.Error())
+	}
 
 	// return err in case of authorization permission mismatch
 	if err != nil && err == syscall.EACCES {
 		return nil, err
 	}
```

Comment:
Since our only opportunity to return errors like `EACCES` is in response to `OpenFile`, this code should probably move there.
Any other code that returns an error should move there too, if it can (if it doesn't require the file to be downloaded first).